examples/triage/baseaddress.cpp

```diff
@@ -0,0 +1,427 @@
+	}
+
+	BNBaseAddressDetectionPOISetting poiSetting = BaseAddressDetectionPOISettingFromString(
+		m_inputs->POIBox->currentText().toStdString());
+	BinaryNinja::BaseAddressDetectionSettings settings = {
+		m_inputs->ArchitectureBox->currentText().toStdString(),
+		m_inputs->AnalysisBox->currentText().toStdString(),
+		minStrlen,
+		alignment,
+		lowerBoundary,
+		upperBoundary,
+		poiSetting,
+		maxPointersPerCluster,
+	};
+
+	if (!m_baseDetection->DetectBaseAddress(settings))
+		emit ResultReady(results);
+
+	auto scores = m_baseDetection->GetScores(&results.Confidence, &results.LastTestedBaseAddress);
+	results.Scores = scores;
+	for (const auto& score : scores)
+	{
+		auto reasons = m_baseDetection->GetReasonsForBaseAddress(score.second);
+		results.Reasons[score.second] = reasons;
+	}
+
+	emit ResultReady(results);
+}
+
+
+void BaseAddressDetectionWidget::HideResultsWidgets(bool hide)
+{
+	if (hide)
+	{
+		m_preferredBaseLabel->setHidden(true);
+		m_preferredBase->setHidden(true);
+		m_confidenceLabel->setHidden(true);
+		m_confidence->setHidden(true);
+		m_resultsTableWidget->setHidden(true);
+		m_reloadBase->setHidden(true);
+		m_rebaseButton->setHidden(true);
+	}
+	else
+	{
+		m_preferredBaseLabel->setHidden(false);
+		m_preferredBase->setHidden(false);
+		m_confidenceLabel->setHidden(false);
+		m_confidence->setHidden(false);
+		m_resultsTableWidget->setHidden(false);
+		m_reloadBase->setHidden(false);
+		m_rebaseButton->setHidden(false);
+	}
+}
+
+
+void BaseAddressDetectionWidget::HandleResults(const BaseAddressDetectionQtResults& results)
+{
+	if (!results.Status.empty())
+		m_status->setText(QString::fromStdString(results.Status));
+
+	if (results.Status.empty() && m_worker->IsAborted())
+		m_status->setText(QString("Aborted by user (Last Base: 0x%1)").arg(
+			QString::number(results.LastTestedBaseAddress, 16)));
+
+	if (results.Scores.empty())
+	{
+		if (!m_worker->IsAborted() && results.Status.empty())
+			m_status->setText("Completed with no results");
+		m_preferredBase->setText("Not available");
+		m_confidence->setText("Not available");
+	}
+	else
+	{
+		HideResultsWidgets(false);
+		if (results.Status.empty() && !m_worker->IsAborted())
+			m_status->setText("Completed with results");
+		m_preferredBase->setText(QString("0x%1").arg(QString::number(results.Scores.rbegin()->second, 16)));
+		m_confidence->setText(QString("%1 (Score: %2)").arg(
+			QString::fromStdString(BaseAddressDetectionConfidenceToString(results.Confidence)),
+			QString::number(results.Scores.rbegin()->first)));
+		m_reloadBase->setText(QString("0x%1").arg(QString::number(results.Scores.rbegin()->second, 16)));
+	}
+
+	m_resultsTableWidget->clearContents();
+	size_t numRows = 0;
+	for (auto rit = results.Scores.rbegin(); rit != results.Scores.rend(); rit++)
+		numRows += results.Reasons.at(rit->second).size();
+
+	m_resultsTableWidget->setRowCount(numRows);
+	size_t row = 0;
+	for (auto rit = results.Scores.rbegin(); rit != results.Scores.rend(); rit++)
```

Comment:
nit: Better - `for(auto & [score, baseaddr] : results.Scores)`

Reply:
Yeah of course...but you know c++20 added: https://en.cppreference.com/w/cpp/ranges/reverse_view